st2api/st2api/controllers/v1/packviews.py

```diff
@@ -0,0 +1,119 @@
+from st2common.persistence.pack import Pack
+from st2common.content.utils import get_pack_file_abs_path
+
+http_client = six.moves.http_client
+
+LOG = logging.getLogger(__name__)
+
+
+class BaseFileController(resource.ResourceController):
+    model = PackAPI
+    access = Pack
+
+    supported_filters = {}
+
+    @jsexpose()
+    def get_all(self, **kwargs):
+        return abort(404)
+
+    def _get_file_content(self, file_path):
+        with open(file_path, 'r') as fp:
+            content = fp.read()
+
+        return content
+
+
+class FilesController(BaseFileController):
+    """
+    Controller which allows user to retrieve content of all the files inside the pack.
+    """
+
+    @jsexpose(arg_types=[str], status_code=http_client.OK)
+    def get_one(self, name_or_id):
+        """
+            Outputs the content of all the files inside the pack.
+
+            Handles requests:
+                GET /packs/views/files/<pack_name>
+        """
+        pack_db = self._get_by_name_or_id(name_or_id=name_or_id)
+        pack_name = pack_db.name
+        pack_files = pack_db.files
+
+        result = []
+        for file_path in pack_files:
+            normalized_file_path = get_pack_file_abs_path(pack_name=pack_name, file_path=file_path)
+
+            if not normalized_file_path or not os.path.isfile(normalized_file_path):
+                # Ignore references to files which don't exist on disk
+                continue
+
+            content = self._get_file_content(file_path=normalized_file_path)
+            item = {
+                'file_path': file_path,
+                'content': content
+            }
+            result.append(item)
+
+        return result
+
+
+class FileController(BaseFileController):
+    """
+    Controller which allows user to retrieve content of a specific file in a pack.
+    """
+    @jsexpose(content_type='text/plain', status_code=http_client.OK)
+    def get_one(self, name_or_id, *file_path_components):
+        """
+            Outputs the content of all the files inside the pack.
+
+            Handles requests:
+                GET /packs/views/files/<pack_name>/<file path>
+        """
+        pack_db = self._get_by_name_or_id(name_or_id=name_or_id)
+
+        if not file_path_components:
+            raise ValueError('Missing file path')
+
+        file_path = os.path.join(*file_path_components)
+        pack_name = pack_db.name
+
+        normalized_file_path = get_pack_file_abs_path(pack_name=pack_name, file_path=file_path)
+
+        if not normalized_file_path or not os.path.isfile(normalized_file_path):
+            # Ignore references to files which don't exist on disk
+            raise StackStormDBObjectNotFoundError('File "%s" not found' % (file_path))
+
+        content = self._get_file_content(file_path=normalized_file_path)
+        return content
+
+
+class PackViewsController(RestController):
+    files = FilesController()
+    file = FileController()
```

Comment:
What I mean is that we have `file` and `files` and it's quite easy to miss or mistype. Better if we would have a single controller that would output either a json list of all files or a content of a single file depending on a presence of `file_path_components`. I get that it is not possible at the moment, I'm just stating that it would result in debugging problems for someone later.